### PR TITLE
Filter Task

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
**Description of change**

- Implement functionality to filter tasks by their status (active or completed), priority, and due date.

**Associated ticket link**
https://trello.com/c/Ksx1iC8m

**Screenshot / Screen Recording**

https://github.com/akib-raihan-sazim/Fancy_To_Do/assets/174452613/d345589d-4bd1-43af-8932-6573e204794e

### CODE REVIEW

File: src/contexts/TasksContext.tsx [hunk 1]
```diff
@@ -66,10 +69,22 @@ const sortTasks = (tasks: Task[]): Task[] => {
   });
 };
 
+const filterTasks = (tasks: Task[], filterStatus: string, filterPriority: string, filterDueDate: Date | null): Task[] => {
+  return tasks.filter(task => {
+    const statusMatch = filterStatus === 'all' || (filterStatus === 'active' ? !task.completed : task.completed);
```
Comment (ishmam-iqbal-sazim) [inline src/contexts/TasksContext.tsx:74]: Do you need the ternary operator here? filterStatus should return true or false anyway.

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comment

File: src/contexts/TasksContext.tsx [hunk 2]
```diff
@@ -66,10 +69,22 @@ const sortTasks = (tasks: Task[]): Task[] => {
...
+    const priorityMatch = filterPriority === 'all' || task.priority === filterPriority;
+    const dueDateMatch = filterDueDate === null || (task.dueDate && task.dueDate.toDateString() === filterDueDate.toDateString());
+    return statusMatch && priorityMatch && dueDateMatch;
```
Comment (ishmam-iqbal-sazim) [inline src/contexts/TasksContext.tsx:77]: Add line break for better readability

File: src/contexts/TasksContext.tsx (hunk 1 above)
Comment (akib-raihan-sazim) [reply src/contexts/TasksContext.tsx:74]: const statusMatch = filterStatus === 'all' || (filterStatus === 'active' && !task.completed) || (filterStatus === 'completed' && task.completed);

I could just write this, as I need to filter between 3 categories ("All" , "Active" , "Completed")

Comment (ishmam-iqbal-sazim) [reply src/contexts/TasksContext.tsx:74]: Okay I see. No I guess keep it as is for now

**Pull request merged**